x/erc20/keeper/ibc_callbacks_test.go

```diff
@@ -0,0 +1,349 @@
+package keeper_test
+
+import (
+	"fmt"
+	"math/big"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/evmos/ethermint/crypto/ethsecp256k1"
+	"github.com/evmos/evmos/v9/testutil"
+	
+
+	transfertypes "github.com/cosmos/ibc-go/v5/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v5/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v5/modules/core/04-channel/types"
+	erc20types "github.com/evmos/evmos/v9/x/erc20/types"
+	ibcgotesting "github.com/cosmos/ibc-go/v5/testing"
+	ibcmock "github.com/cosmos/ibc-go/v5/testing/mock"
+
+	claimstypes "github.com/evmos/evmos/v9/x/claims/types"
+	"github.com/evmos/evmos/v9/x/erc20/keeper"
+	"github.com/evmos/evmos/v9/x/erc20/types"
+	"github.com/evmos/evmos/v9/contracts"
+	vestingtypes "github.com/evmos/evmos/v9/x/vesting/types"
+)
+
+var (
+	erc20Denom   = "erc20/0xdac17f958d2ee523a2206206994597c13d831ec7"
+)
+
+func (suite *KeeperTestSuite) TestOnRecvPacket() {
+	// secp256k1 account
+	secpPk := secp256k1.GenPrivKey()
+	secpAddr := sdk.AccAddress(secpPk.PubKey().Address())
+	secpAddrEvmos := secpAddr.String()
+	secpAddrCosmos := sdk.MustBech32ifyAddressBytes(sdk.Bech32MainPrefix, secpAddr)
+
+	// ethsecp256k1 account
+	ethPk, err := ethsecp256k1.GenerateKey()
+	suite.Require().Nil(err)
+	ethsecpAddr := sdk.AccAddress(ethPk.PubKey().Address())
+	ethsecpAddrEvmos := sdk.AccAddress(ethPk.PubKey().Address()).String()
+	ethsecpAddrCosmos := sdk.MustBech32ifyAddressBytes(sdk.Bech32MainPrefix, ethsecpAddr)
+
+	// Setup Cosmos <=> Evmos IBC relayer
+	denom := "uatom"
+	sourceChannel := "channel-292"
+	evmosChannel := claimstypes.DefaultAuthorizedChannels[1]
+	path := fmt.Sprintf("%s/%s", transfertypes.PortID, evmosChannel)
+
+	timeoutHeight := clienttypes.NewHeight(0, 100)
+	disabledTimeoutTimestamp := uint64(0)
+	mockPacket := channeltypes.NewPacket(ibcgotesting.MockPacketData, 1, transfertypes.PortID, "channel-0", transfertypes.PortID, "channel-0", timeoutHeight, disabledTimeoutTimestamp)
+	packet := mockPacket
+	expAck := ibcmock.MockAcknowledgement
+
+	coins := sdk.NewCoins(
+		sdk.NewCoin("aevmos", sdk.NewInt(1000)),
+		sdk.NewCoin(erc20Denom, sdk.NewInt(1000)), // some ERC20 token
+		sdk.NewCoin(cosmosTokenBase, sdk.NewInt(1000)), // some coin with a registered token pair
+	)
+
+	testCases := []struct {
+		name        string
+		malleate    func()
+		ackSuccess  bool
+		ibcConv     bool
```

Comment:
nit, might be more clear to not mention ibc here, since this flag just says if conversion is expected or not. 
```suggestion
		expConversion bool
```